Warn AMO users when visiting the site with Fenix

bobsilverberg opened the pull request:
Fixes #9068 

This PR introduces a few changes:

1. The `AddonCompatibilityError`, which is the red warning near the top of the add-on detail card, will no longer be displayed when a user visits the site with Fenix. This is because that warning is being replaced by the `WrongPlatformWarning`. Before screenshot:

![Screenshot 2020-01-13 12 09 58](https://user-images.githubusercontent.com/142755/72276329-acc7e180-35fd-11ea-9227-87b160a90961.png)

2. When visiting the site with Fenix, the `WrongPlatformWarning` will be displayed with a message. This will either be displayed under the install button (on the add-on detail page):

![Screenshot 2020-01-13 12 07 39](https://user-images.githubusercontent.com/142755/72276219-74280800-35fd-11ea-9cd7-a805e8ef19fd.png)

or at the top of the screen:

![Screenshot 2020-01-13 12 08 00](https://user-images.githubusercontent.com/142755/72276239-8144f700-35fd-11ea-9668-a886dbd7530c.png)

3. Similar to 2, above, the standard `WrongPlatformWarning`, which displays a warning if you visit the `/android` site with desktop, or the `/firefox` site with Android, will *not* be displayed for Fenix users, as it will be replaced by the new `WrongPlatformWarning` as described in 2, above.

bobsilverberg commented:
Well, I tried to use mozilla/addons-engineering for this assignment, but it just chose @willdurand anyway. 😜

bobsilverberg commented:
Note that we are waiting on a final SUMO link for this, so we should not merge this until we have that. I will update the milestone if we don't have a proper link by tomorrow.

bobsilverberg commented:
@caitmuenster Please let me know once we have a correct link for SUMO for this.

willdurand commented on src/amo/components/WrongPlatformWarning/index.js:
```diff
@@ -35,46 +43,61 @@ type InternalProps = {|
+      message = i18n.sprintf(
+        this.props.fixFenixLinkMessage ||
+          i18n.gettext(
+            `To find add-ons compatible with Firefox on Android,
```
I think it is `Firefox for Android`

bobsilverberg replied:
Indeed! That's also what's in the doc at https://docs.google.com/document/d/1HOImPu2RO9PXIbrNvE_YhYyONw7n-rgjzMtMQ3Knk4Q/edit?ts=5e18f98a#heading=h.lzrviwt2ql63. I have no idea where that "on" came from!

caitmuenster commented:
Thanks @bobsilverberg! The article is published at https://support.mozilla.org/kb/add-compatibility-firefox-preview/.

bobsilverberg commented:
Thanks @willdurand and @caitmuenster. I addressed the review comment and have updated the SUMO link. I still have an open question for @jvillalobos at https://github.com/mozilla/addons-frontend/issues/9068#issuecomment-574193847 about the wording for the warnings, but that could be a follow-up.
